### ISSUE TITLE
ZFS does not support `RENAME_NOREPLACE` flag on `renameat2`

### ISSUE THREAD

**Issue by ReactorScram** [untitaker/rust-atomicwrites#63]
Had a user hit this on the Firezone Client https://github.com/firezone/firezone/issues/6704#issuecomment-2355357400

This repo replicates the issue when run on a ZFS system https://github.com/ReactorScram/atomicwrites-zfs-repro

For now I'll work around this by either wrapping `atomicwrites` with something that handles the error and falls back on a simple rename, or I'll switch to SQLite since I have a few files I'm writing and it's getting to the point where a DB would be handy.

**Comment by sunfishcode** [comment]
rust-atomicwrites does have fallback code for when `renameat2` isn't available, but it isn't catching this case because it's only checking for `ENOSYS`, and the `renameat2` here is failing with `EINVAL`, presumably due to the unsupported flag.

From the strace log in the original report:
```
renameat2(11, "tmpfile.tmp", 12, "firezone-id.json", RENAME_NOREPLACE) = -1 EINVAL (Invalid argument)
```

I've now filed #64 with a fix.

**Comment by sunfishcode** [comment]
This is now fixed in atomicwrites 0.4.4.
